htdocs/contrat/class/api_contracts.class.php

```diff
@@ -208,8 +209,15 @@ public function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100,
 	public function post($request_data = null)
 	{
 		if (!DolibarrApiAccess::$user->hasRight('contrat', 'creer')) {
-			throw new RestException(403, "Insufficient rights");
+			throw new RestException(403, "Missing permission: Create/modify contracts/subscriptions");
 		}
+
+		$socid = (int) $request_data['socid'];
+		$thirdparties = new Thirdparties();
+		$thirdparty_result = $thirdparties->get((int) $socid);
```

Comment:
Using the get of another api is doing a different job that what we want. We just want to check if thirdparty exists.
For this, a standard :
```
$thirdpartytmp = new Societe($db);
$thirdparty_result = $thirdpartytmp>fetch($socid);
if ($thirdparty_result <= 0) {
    throw new RestException(404, 'Thirdparty not found or not allowed');
}
```
is enough (we don't need to load the discounts, check the massemailing status, ... done by the get of API).
If use with permission to create contract has no permission on thirdparties, he should not have more detail if thirdparty exists or not (only the one who haspermission to read thirdparties is allowed to get this information). This is another reason why we should not use the getofapi but the get of thirdparty here.
It is samefor the put.